### PR TITLE
EditableHeading -Add option to pass argument to onStartEditing, input class name prop,…

### PR DESCRIPTION
**Pull request by mayaAssayag**
editable heading component: add an option to pass an argument with onStartEditing prop, input class prop, data-testid prop.

### CODE REVIEW

File: src/components/EditableHeading/EditableHeading.jsx [hunk 1]
```diff
@@ -139,7 +141,7 @@ const EditableHeading = props => {
     const inputType = props.inputType || textAreaType;
     return {
       value: valueState,
-      className: `editable-heading-input element-type-${props.type} size-${props.size}`,
+      className: `editable-heading-input element-type-${props.type} size-${props.size} ${inputClassName}`,
```
Comment (orrgottlieb) [inline src/components/EditableHeading/EditableHeading.jsx:144]: please please use `cx` something like - 
`cx(editable-heading-input element-type-${props.type}, editable-heading-input element-type-${props.type}, inputClassName)`

File: src/components/EditableHeading/EditableHeading.jsx [hunk 2]
```diff
@@ -246,7 +261,10 @@ EditableHeading.defaultProps = {
   ignoreBlurClass: undefined,
   onFinishEditing: undefined,
   onCancelEditing: undefined,
-  onIgnoreBlurEvent: undefined
+  onIgnoreBlurEvent: undefined,
+  style: undefined,
+  dataTestId: "",
```
Comment (orrgottlieb) [inline src/components/EditableHeading/EditableHeading.jsx:266]: add a default please

File: src/components/EditableHeading/EditableHeading.jsx [hunk 3]
```diff
@@ -246,7 +261,10 @@ EditableHeading.defaultProps = {
   ignoreBlurClass: undefined,
   onFinishEditing: undefined,
   onCancelEditing: undefined,
-  onIgnoreBlurEvent: undefined
+  onIgnoreBlurEvent: undefined,
+  style: undefined,
```
Comment (orrgottlieb) [inline src/components/EditableHeading/EditableHeading.jsx:265]: did you use this props? if not can you please remove it

File: src/components/EditableHeading/__tests__/editableHeading.jest.js [hunk 4]
```diff
@@ -38,4 +37,40 @@ describe("EditableHeading", () => {
 
     expect(button).toBeInTheDocument();
   });
+
+  it("renders correctly with dataTestId prop", () => {
```
Comment (orrgottlieb) [inline src/components/EditableHeading/__tests__/editableHeading.jest.js:41]: like!!!!

File: src/components/EditableHeading/__tests__/editableHeading.jest.js [hunk 5]
```diff
@@ -38,4 +37,40 @@ describe("EditableHeading", () => {
...
+    const mockCallback = jest.fn();
+    const { getByRole } = render(<EditableHeading onStartEditing={mockCallback} />);
+
+    const input = getByRole("input");
+    fireEvent.click(input);
+
+    expect(mockCallback.mock.calls.length).toBe(1);
+  });
+  
+  it("should play the onStartEditing function with params when click on the input element inside the EditableHeading element", () => {
```
Comment (orrgottlieb) [inline src/components/EditableHeading/__tests__/editableHeading.jest.js:67]: cool!

File: src/components/EditableHeading/__tests__/editableHeading.jest.js [hunk 6]
```diff
@@ -38,4 +37,40 @@ describe("EditableHeading", () => {
...
+
+  it("renders correctly with dataTestId prop", () => {
+    const { getByTestId } = render(<EditableHeading dataTestId="data-test-id" />);
+
+    const editableHeadingElement = getByTestId("data-test-id");
+
+    expect(editableHeadingElement).toBeInTheDocument();
+  });
+
+  it("renders correctly with dataTestId prop", () => {
```
Comment (orrgottlieb) [inline src/components/EditableHeading/__tests__/editableHeading.jest.js:49]: Nice!

**Review by orrgottlieb — CHANGES_REQUESTED** [review]

File: src/components/EditableHeading/EditableHeading.jsx (hunk 1 above)
Comment (mayaAssayag) [reply src/components/EditableHeading/EditableHeading.jsx:144]: fixed

File: src/components/EditableHeading/EditableHeading.jsx (hunk 3 above)
Comment (mayaAssayag) [reply src/components/EditableHeading/EditableHeading.jsx:265]: I'm not using them, but how can we know that others have not used them already.
or do you mean just remove them from the storybook description?

File: src/components/EditableHeading/EditableHeading.jsx (hunk 2 above)
Comment (orrgottlieb) [reply src/components/EditableHeading/EditableHeading.jsx:275]: @mayaAssayag 
According to react it should either be undefined or CSSObject type

File: src/components/EditableHeading/EditableHeading.jsx (hunk 3 above)
Comment (orrgottlieb) [reply src/components/EditableHeading/EditableHeading.jsx:274]: sorry, i didn't see that we already used it before

**Review by orrgottlieb — APPROVED** [review]

**Review by orrgottlieb — APPROVED** [review]

**Pull request merged**